packages/application-extension/src/index.ts

```diff
@@ -178,9 +180,11 @@ const opener: JupyterFrontEndPlugin<void> = {
 
         const file = decodeURIComponent(path);
         const ext = PathExt.extname(file);
-        app.restored.then(() => {
+        app.restored.then(async () => {
           // TODO: get factory from file type instead?
           if (ext === '.ipynb') {
+            await settingRegistry?.load('@jupyterlab/notebook-extension:panel');
```

Comment:
The same trick was added on the `main` branch (targeting lab 4.0) in https://github.com/jupyterlab/retrolab/pull/319.

However for RetroLab `0.3.x` we should find a better way to handle this.